api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -243,4 +241,132 @@ public Bed getLatestBedByVisit(String visitUuid) {
                 .setParameter("voided", false)
                 .list();
     }
+
+    @Override
+    public List<AdmissionLocation> getAdmissionLocations() {
+        List<Integer> admissionLocationIds = getAdmissionLocationIds();
```

Comment:
@sanishmaharjan the original logic (see the master branch) seemed better. There was a private method to list locations given a tag, why did you feel this should be changed?

Importantly I am not fond of `getAdmissionLocationIds()` because it is a **public** method that exposes database IDs.
Try to revert to a reusable method like the original that was returning a list of locations given any tag.

Reply:
And while doing so, could you investigate why [this](https://github.com/openmrs/openmrs-core/blob/91aecd7822776ade37b06e906766a139b6b68189/api/src/main/java/org/openmrs/api/LocationService.java#L179) wasn't used?

So basically compare [this in Core](https://github.com/openmrs/openmrs-core/blob/91aecd7822776ade37b06e906766a139b6b68189/api/src/main/java/org/openmrs/api/impl/LocationServiceImpl.java#L189-L201) and [this on the master branch](https://github.com/openmrs/openmrs-module-bedmanagement/blob/efe8559f76ad7470efb4c74763432e6134450c84/api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java#L69-L72), and if there is not any good reason to avoid the Core method, then use it.

Reply:
@mks-d  getAdmissionLocationIds() has been changed to getLocationsByTagName(String tagName).
I didn't use getPhysicalLocationsByLocationTag(String locationTagName, Session session) since we need all location with tag "Admission Location" But that function return only child admission location with tag "Admission Location".

Reply:
Ok `getAdmissionLocations()` at the DAO layer should take  a `List<Location>` as argument. And `getAdmissionLocations()` at the service layer should fetch the locations tagged as 'admission' using the Core service and then pass that list to the function at the DAO layer.

Reply:
Done.